vimc-4136 Add get run status for polling running reports

EmmaLRussell opened the pull request:
Also added unit tests for the api class (only had integration tests before), and a custom error class for response errors from OrderlyWeb.

EmmaLRussell committed:
Added status poll method

EmmaLRussell committed:
Added tests for result class

EmmaLRussell committed:
Added api unit tests

EmmaLRussell committed:
Use custom error class

EmmaLRussell committed:
Add new classes to package

EmmaLRussell committed:
style fix

hillalex commented on orderlyweb_api/orderly_web_api.py:
```diff
@@ -9,21 +11,40 @@ def __init__(self, base_url, token):
         if response.status_code != 200:
             msg = 'Unexpected status code: {}. Unable to authenticate.'
-            raise Exception(msg.format(response.status_code))
+            raise OrderlyWebResponseError(msg.format(response.status_code),
```
Might it be more useful to just pass through the error message from orderly web? These should be readable, informative things like "Invalid token" or "Missing required permissions: ..."

hillalex commented on orderlyweb_api/orderly_web_api.py:
```diff
@@ -9,21 +11,40 @@ def __init__(self, base_url, token):
+    @staticmethod
+    def handle_response(response):
+        if response.status_code != 200:
+            msg = 'Unexpected status code: {}'
```
Similarly here, why obscure the original error message?

hillalex commented on orderlyweb_api/orderly_web_api.py:
```diff
@@ -9,21 +11,40 @@ def __init__(self, base_url, token):
-            raise Exception(msg.format(response.status_code))
+            raise OrderlyWebResponseError(msg.format(response.status_code),
+                                          response=response)
         self.token = response.json()['access_token']
```
So `self.token` is an OW access token, and the `token` that is passed to `__init__` is a Montagu token? Perhaps relname one or both to make this super clear.

hillalex reviewed (commented):
Looks good to me, couple of small suggestions.

EmmaLRussell committed:
Extract error message from orderly web response and rename tokens in api class

EmmaLRussell committed:
user orderlyweb deploy

EmmaLRussell committed:
Move here var

EmmaLRussell commented:
I've updated the error class to extract error message from the orderly web response, and I changed the scripts to use orderyweb deploy while I was in there.

hillalex commented on orderlyweb_api/orderly_web_api.py:
```diff
@@ -4,16 +4,14 @@
 
 class OrderlyWebAPI:
-    def __init__(self, base_url, token):
+    def __init__(self, base_url, auth_token):
```
Sorry to be a pain, but I still don't find `auth_token` and `access_token` to be transparently different terms - can this one be `montagu_token`?

EmmaLRussell replied:
It could theoretically be a Github token though, couldn't it?

EmmaLRussell replied:
Would something like `auth_provider_token` be better?

hillalex replied:
Ah of course, sorry I completely forgot about that! `auth_provider_token` would be perfect

EmmaLRussell committed:
Rename auth_token to auth_provider_token
